packages/extension-ui/src/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019-2020 @polkadot/extension-ui authors & contributor
+// SPDX-License-Identifier: Apache-2.0
+
+import { WithTranslation } from 'react-i18next';
+
+import React from 'react';
+// import styled from 'styled-components';
+
+import translate from './translate';
+import Header from '../partials/Header';
+import ActionText from './ActionText';
+// import { ThemeProps } from '../types';
+
+interface Props extends WithTranslation {
+  children: React.ReactNode;
+  className?: string;
+  error?: Error | null;
+  onError?: () => void;
+  trigger?: unknown;
+}
+
+interface State {
+  error: Error | null;
+  prevTrigger: string | null;
+}
+
+// NOTE: This is the only way to do an error boundary, via extend
+class ErrorBoundary extends React.Component<Props> {
+  state: State = { error: null, prevTrigger: null };
+
+  static getDerivedStateFromError (error: Error): Partial<State> {
+    return { error };
+  }
+
+  static getDerivedStateFromProps ({ trigger }: Props, { prevTrigger }: State): State | null {
+    const newTrigger = JSON.stringify({ trigger });
+
+    return (prevTrigger !== newTrigger)
+      ? { error: null, prevTrigger: newTrigger }
+      : null;
+  }
```

Comment:
Can be removed now?

Reply:
I've been scratching my head a lot about this yesterday simply because I don't really understand it. The implementation I used before is also using some kind of comparison ([here](https://github.com/bvaughn/react-error-boundary/blob/master/src/index.js#L3-L4)) that I don't really understand.